src/Umbraco.Web/Compose/NestedContentPropertyComponent.cs

```diff
@@ -0,0 +1,180 @@
+                foreach (var nestedContentProp in nestedContentProps)
+                {
+                    // A NC Prop may have one or more values due to cultures
+                    var propVals = nestedContentProp.Values;
+                    foreach (var cultureVal in propVals)
+                    {
+                        // Remove keys from published value & any nested NC's
+                        var updatedPublishedVal = CreateMissingNestedContentKeys(cultureVal.PublishedValue?.ToString());
+                        cultureVal.PublishedValue = updatedPublishedVal;
+
+                        // Remove keys from edited/draft value & any nested NC's
+                        var updatedEditedVal = CreateMissingNestedContentKeys(cultureVal.EditedValue?.ToString());
+                        cultureVal.EditedValue = updatedEditedVal;
+                    }
+                }
+            }
+        }
+
+        public void Terminate()
+        {
+            ContentService.Copying -= ContentService_Copying;
+            ContentService.Saving -= ContentService_Saving;
+            ContentService.Publishing -= ContentService_Publishing;
+        }
+
+        private string CreateNewNestedContentKeys(string ncJson)
+        {
+            if (string.IsNullOrWhiteSpace(ncJson))
+                return ncJson;
+
+            // Try & convert JSON to JArray (two props we will know should exist are key & ncContentTypeAlias)
+            var ncItems = JArray.Parse(ncJson);
+
+            // NC prop contains one or more items/rows of things
+            foreach (var nestedContentItem in ncItems.Children<JObject>())
+            {
+                foreach (var ncItemProp in nestedContentItem.Properties())
+                {
+                    if(ncItemProp.Name.ToLowerInvariant() == "key")
+                        ncItemProp.Value = Guid.NewGuid().ToString();
+
+                    // No need to check this property for JSON - as this is a JSON prop we know
+                    // That onyl contains the string of the doctype alias used as the NC item
+                    if (ncItemProp.Name == NestedContentPropertyEditor.ContentTypeAliasPropertyKey)
+                        continue;
+
+                    // As we don't know what properties in the JSON may contain the nested NC
+                    // We are detecting if its value stores JSON to help filter the list AND that in its JSON it has ncContentTypeAlias prop
+                    if (ncItemProp.Value.ToString().DetectIsJson() && ncItemProp.Value.ToString().Contains(NestedContentPropertyEditor.ContentTypeAliasPropertyKey))
+                    {
+                        // Recurse & update this JSON property
+                        ncItemProp.Value = CreateNewNestedContentKeys(ncItemProp.Value.ToString());
+                    }
+                }
+            }
+
+            return ncItems.ToString();
+        }
+
+        private string CreateMissingNestedContentKeys(string ncJson)
```

Comment:
This is almost a total duplication of the above :/

Reply:
Fixed duplication, the method takes in a bool to know if it's replacing the key values OR only updating missing key properties to have a new key